gammapy/maps/wcsnd.py

```diff
@@ -231,10 +231,27 @@ def set_by_idx(self, idx, vals):
         idx = pix_tuple_to_idx(idx)
         self.data.T[idx] = vals
 
-    def sum_over_axes(self):
+    def sum_over_axes(self, keepdims=False):
+        """To sum map values over all non-spatial axes.
+
+        Parameters
+        ----------
+        keepdims : bool, optional
+            If this is set to true, the axes which are summed over are left in
+            the map with a single bin
+
+        Returns
+        -------
+        map_out : WcsNDMap
+            Map with non-spatial axes summed over
+        """
         axis = tuple(range(self.data.ndim - 2))
-        data = np.nansum(self.data, axis=axis)
+        data = np.nansum(self.data, axis=axis, keepdims=keepdims)
         geom = self.geom.to_image()
```

Comment:
Maybe move this line to an `else` block below? I think this improves the code structure a bit.

Reply:
I think an `else` block is not necessary here. 
But I have moved the `data = np.nansum(self.data, axis=axis, keepdims=keepdims)` to after the `if` block, which does make the code clearer.